### DISCUSSION TITLE
Conversion between spectral library formats

### DISCUSSION THREAD

**Discussion by hguturu** [vdemichev/DiaNN#470] [Q&A]
Continuing from the discussion at https://github.com/vdemichev/DiaNN/discussions/74.

Using the interface the command generated is `diann.exe --lib report-lib.predicted.speclib --threads 8 --verbose 1 --out report.tsv --qvalue 0.01 --matrices --out-lib diann_v1.8.1_default.tsv --gen-spec-lib --reanalyse --relaxed-prot-inf --smart-profiling --peak-center --no-ifs-removal`. 

I found this is equivalent to running `C:\DIA-NN\1.8.1\DiaNN.exe --lib report-lib.predicted.speclib --threads 8 --verbose 1 --out report.tsv --out-lib diann_v1.8.1_cmd.tsv --gen-spec-lib`.

i.e., Besides `--gen-spec-lib` all the other flags where unnecessary. Which makes sense since no MS files are being analyzed.

But, I went ahead and kept using the interface to be safe and one item I am interested in the round trim .speclib -> .tsv -> .speclib (since speclibs are smaller and more performant). 

I noticed the round trip works, but creates two files.

For example, I was able to export the library from `.speclib` to `.tsv` using the interface (logged command):
`diann.exe --lib report-lib.predicted.speclib --threads 8 --verbose 1 --out report.tsv --qvalue 0.01 --matrices --out-lib diann_v1.8.1_default.tsv --gen-spec-lib --reanalyse --relaxed-prot-inf --smart-profiling --peak-center --no-ifs-removal`

And when try to convert back to speclib (to ensure I can go back after modifying the .tsv), I get two files. Again ran it via the interface (logged command):
`diann.exe --lib "D:\ref_vep\diann_v1.8.1_default.tsv" --threads 8 --verbose 1 --out "C:\DIA-NN\1.8.1\report.tsv" --qvalue 0.01 --matrices  --out-lib "D:\ref_vep\diann_v1.8.1_default.round_trip.speclib" --gen-spec-lib --reanalyse --relaxed-prot-inf --smart-profiling --peak-center --no-ifs-removal`

I get `D:\ref_vep\diann_v1.8.1_default.round_trip.speclib` and I get `D:\ref_vep\diann_v1.8.1_default.speclib`. `D:\ref_vep\diann_v1.8.1_default.round_trip.speclib` is massive (similar size to .tsv) and perhaps an uncompressed intermediate of `D:\ref_vep\diann_v1.8.1_default.speclib`, but then the `D:\ref_vep\diann_v1.8.1_default.round_trip.speclib` is not cleaned up. I assume `D:\ref_vep\diann_v1.8.1_default.speclib` is all that is needed, but it would be nice if the final file was `D:\ref_vep\diann_v1.8.1_default.round_trip.speclib` as required by the `--out-lib` flag. Unless I am missing something and need both files.

Size In bytes:
```
-a---         8/18/2022   4:56 PM     1500415549 diann_v1.8.1_default.tsv.speclib
-a---         8/18/2022   5:49 PM   131773480207 diann_v1.8.1_default.round_trip.speclib
-a---         7/21/2022   6:37 PM     1500415691 report-lib.predicted.speclib
```

I also noticed the round trip has slightly different protein group (off by one), but this might be minor and non issue..
e.g. speclib -> tsv, the output is 
`[0:11] Spectral library loaded: 130480 protein isoforms, 150018 protein groups and 4897554 precursors in 1525554 elution groups.`

but, tsv -> speclib, the output is
`[20:24] Spectral library loaded: 130480 protein isoforms, 150017 protein groups and 4897554 precursors in 1525554 elution groups.`

**Comment by vdemichev** [comment]
--out-lib can only specify .tsv files. So the D:\ref_vep\diann_v1.8.1_default.round_trip.speclib file is actually a tab-separated text file. The D:\ref_vep\diann_v1.8.1_default.speclib is a proper speclib.
